crates/network-scanner/src/interfaces/filter.rs

```diff
@@ -0,0 +1,42 @@
+use std::net::IpAddr;
+
+use super::NetworkInterface;
+
+#[derive(Debug, Clone)]
+pub struct Filter {
+    pub ignore_ipv6: bool,
+    pub include_loopback: bool,
+}
```

Comment:
issue: I think naming in this module should be improved. Or rather, it seems that the `Filter` struct is trying to achieve two different goals at once, and that’s why it resulted into a confusing naming. For instance, I didn’t expect `Filter::matches` to be used specifically for excluding loopback interfaces, and `Filter::clear` should be named something else, because `clear` is typically used to empty collections, and at least it’s not clear what it means for a "filter".

It may be simpler to use two free-standing functions instead.

Or maybe you want to keep `Filter` to use to pass the filter parameters to `get_network_interface`, but extract the parameters and pass them to functions instead of declaring the operations on `Filter`.